scrapi/base/helpers.py

```diff
@@ -118,11 +118,15 @@ def oai_process_uris(*args):
     provider_uris = []
     for item in identifiers:
         if 'doi' in item.lower():
-            doi = item.replace('doi:', '').replace('DOI:', '').strip()
-            if 'http://dx.doi.org/' in doi:
-                object_uris.append(doi)
-            else:
-                object_uris.append('http://dx.doi.org/{}'.format(doi))
+            try:
```

Comment:
The complexity of this function is getting a little high. Might be a good idea to break some of the functionality into smaller helper functions, just to make the flow of logic a little more transparent.